### DISCUSSION TITLE
Newbie Question

### DISCUSSION THREAD

**Discussion by nitintomer** [openai/whisper#1372] [General]
I am a newbie, trying to install Whisper on my Windows 11 laptop.

I did the following -

1. Installed Python.
2. Installed PIP
3. pip install -U openai-whisper
4. scoop install ffmpeg
5. pip install setuptools-rust

I believe all steps are completed, but when I try to run whisper - I got the message "'whisper' is not recognized as an internal or external command,
operable program or batch file.". I know I am doing something stupid - but not able to figure it out. Please help.

Thanks

Nitin

**Comment by phineas-pta** [comment]
when you executed those commands, did you get any error?

**Reply by nitintomer** [reply]
No, all these commands were run successfully.

**Comment by WildBirdTG** [comment]
Assuming your installation are all successful, first thing to check is your PATH variable, it should include something like this:
C:\Users\admin\AppData\Local\Programs\Python\Python310\Scripts\
or similar to your python's installation. If not, put your python installation path and the scripts path into $PATH.

**Reply by nitintomer** [reply]
The PATH variable has "C:\Users\XXXXXX\AppData\Local\Microsoft\WindowsApps\" in it. PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0 is a sub-folder under it. I am able to call python and pip from command line.

**Reply by phineas-pta** [reply]
can you run this command to see where whisper is installed `pip show whisper`

**Reply by nitintomer** [reply]
Looks like it's not installed, I got this - WARNING: Package(s) not found: whisper

**Reply by nitintomer** [reply]
I tried to install again, and got this result -

Requirement already satisfied: openai-whisper in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (20230314)
Requirement already satisfied: numba in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from openai-whisper) (0.57.0)
Requirement already satisfied: numpy in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from openai-whisper) (1.24.3)
Requirement already satisfied: torch in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from openai-whisper) (2.0.1)
Requirement already satisfied: tqdm in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from openai-whisper) (4.65.0)
Requirement already satisfied: more-itertools in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from openai-whisper) (9.1.0)
Requirement already satisfied: tiktoken==0.3.1 in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from openai-whisper) (0.3.1)
Requirement already satisfied: ffmpeg-python==0.2.0 in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from openai-whisper) (0.2.0)
Requirement already satisfied: future in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from ffmpeg-python==0.2.0->openai-whisper) (0.18.3)
Requirement already satisfied: regex>=2022.1.18 in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from tiktoken==0.3.1->openai-whisper) (2023.5.5)
Requirement already satisfied: requests>=2.26.0 in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from tiktoken==0.3.1->openai-whisper) (2.30.0)
Requirement already satisfied: llvmlite<0.41,>=0.40.0dev0 in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from numba->openai-whisper) (0.40.0)
Requirement already satisfied: filelock in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from torch->openai-whisper) (3.12.0)
Requirement already satisfied: typing-extensions in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from torch->openai-whisper) (4.5.0)
Requirement already satisfied: sympy in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from torch->openai-whisper) (1.12)
Requirement already satisfied: networkx in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from torch->openai-whisper) (3.1)
Requirement already satisfied: jinja2 in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from torch->openai-whisper) (3.1.2)
Requirement already satisfied: colorama in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from tqdm->openai-whisper) (0.4.6)
Requirement already satisfied: charset-normalizer<4,>=2 in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from requests>=2.26.0->tiktoken==0.3.1->openai-whisper) (3.1.0)
Requirement already satisfied: idna<4,>=2.5 in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from requests>=2.26.0->tiktoken==0.3.1->openai-whisper) (3.4)
Requirement already satisfied: urllib3<3,>=1.21.1 in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from requests>=2.26.0->tiktoken==0.3.1->openai-whisper) (2.0.2)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from requests>=2.26.0->tiktoken==0.3.1->openai-whisper) (2023.5.7)
Requirement already satisfied: MarkupSafe>=2.0 in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from jinja2->torch->openai-whisper) (2.1.2)
Requirement already satisfied: mpmath>=0.19 in c:\users\XXXX\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from sympy->torch->openai-whisper) (1.3.0)

**Reply by phineas-pta** [reply]
sorry my bad, it should be `pip show openai-whisper`

**Reply by nitintomer** [reply]
Looks like it's there -

Name: openai-whisper
Version: 20230314
Summary: Robust Speech Recognition via Large-Scale Weak Supervision
Home-page: https://github.com/openai/whisper
Author: OpenAI
Author-email:
License: MIT
Location: C:\Users\XXXX\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages
Requires: ffmpeg-python, more-itertools, numba, numpy, tiktoken, torch, tqdm
Required-by:

**Reply by phineas-pta** [reply]
it's certainly a PATH problem, now you just have to find out where is the executable `whisper.exe`

can you check if it's in `...\LocalCache\local-packages\Python311\Scripts` or `Python311\bin` ?

**Reply by nitintomer** [reply]
Thanks, I was able to make it work by adding it to the PATH variable.
